### PR TITLE
Report build scan links

### PR DESCRIPTION
**Pull request by alextu**
This PR introduces another init script to fetch and attach build scan links to jobs.
See the README changes for more details on usage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,58 @@ For all other options see `inputs` section in [develocity-gradle.yml](develocity
...
+
+![links.png](images/links.png)
+
+Using GitLab templating, that can be factored and applied to multiple jobs:
+
+```yml
+.gradle-inject-job:
+    before_script:
+        - !reference [ .injectDevelocityForGradle ]
+        - !reference [ .reportBuildScanLinks ]
```
Comment (welandaz) [inline README.md]: There is no way to bundle them as one script, right?

**Review by welandaz — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (alextu) [reply README.md]: That's a good point, I split it to follow a bit what we already have in the GH action, but it puts some burden on the user side. I'll see if I can arrange something to have best of both worlds: dedicated scripts but only one to reference.

File: README.md [hunk 2]
```diff
@@ -36,6 +36,58 @@ For all other options see `inputs` section in [develocity-gradle.yml](develocity
 
 > **_NOTE:_** The build is also instrumented with our [Common Custom User Data Gradle plugin](https://github.com/gradle/common-custom-user-data-gradle-plugin) as well, as it will provide more details about your build.
 
+#### Build scan links
```
Comment (guylabs) [inline README.md]:
```suggestion
#### Build Scan links
```

File: README.md [hunk 3]
```diff
@@ -36,6 +36,58 @@ For all other options see `inputs` section in [develocity-gradle.yml](develocity
...
+An optional init script can be used to generate a report containing build scan links.
```
Comment (guylabs) [inline README.md]:
```suggestion
An optional init script can be used to generate a report containing Build Scan links.
```

File: README.md [hunk 4]
```diff
@@ -36,6 +36,58 @@ For all other options see `inputs` section in [develocity-gradle.yml](develocity
...
+    - !reference [.reportBuildScanLinks]
+    # Pass it to each Gradle invocation
+    - ./gradlew clean -I $DEVELOCITY_INIT_SCRIPT_PATH -I $DEVELOCITY_REPORT_INIT_SCRIPT_PATH
+    - ./gradlew check -I $DEVELOCITY_INIT_SCRIPT_PATH -I $DEVELOCITY_REPORT_INIT_SCRIPT_PATH
+    # Attach the report
+  artifacts:
+    !reference [ .build_scan_links_report, artifacts ]
+```
+
+This shows the build scan links on the job details right panel:
```
Comment (guylabs) [inline README.md]:
```suggestion
This shows the Build Scan links on the job details right panel:
```

**Review by guylabs — APPROVED** [review]

**Pull request merged**